projects/plugins/jetpack/modules/memberships/class-jetpack-memberships.php

```diff
@@ -454,6 +454,7 @@ public static function user_can_view_post() {
 
 		$newsletter_access_level = self::get_newsletter_access_level();
 		$paywall                 = \Automattic\Jetpack\Extensions\Premium_Content\subscription_service();
+		// TODO: Figure out why the `get_the_ID` is passed here when the function accepts a `$is_blog_subscriber` param
```

Comment:
I think there are 2 issues here: the common signature for  `visitor_can_view_content` is `($valid_plan_ids, $access_level)`, however `WPCOM_Online_Subscription_Service` adds `$is_blog_subscriber` as a third optional parameter.
So this code will fail if the subscription service is different than `WPCOM_Online_Subscription_Service`.

I am failing to understand the exact context of this call, but either:
- we call 
```
new WPCOM_Online_Subscription_Service()->visitor_can_view_content( self::get_all_plans_id_jetpack_recurring_payments(), $newsletter_access_level, true
);
```


Or we call
```
return $paywall->visitor_can_view_content( self::get_all_plans_id_jetpack_recurring_payments(), $newsletter_access_level
);

```
I think we should just remove `get_the_ID()`. This is an artefact of a probable copy/paste error with `user_can_view_content`

Reply:
> I think we should just remove get_the_ID(). This is an artefact of a probable copy/paste error with user_can_view_content

I agree, I was thinking the same. Fixed in [e81e819](https://github.com/Automattic/jetpack/pull/28140/commits/e81e8191db47506f1756550539aa5fcad8389345)